run/o2aod_mc_to_hepmc.cxx

```diff
@@ -53,188 +27,25 @@ void customize(std::vector<ConfigParamSpec>& workflowOptions)
-   *
-   *  @param collision  Event header
-   *  @param tracks     Tracks of the event
-   */
-  void process(Header const& collision,
-               Tracks const& tracks)
-  {
-    LOG(debug) << "=== Processing only tracks ===";
-    mConverter.startEvent();
-    mConverter.process(collision, tracks);
-    mConverter.endEvent();
-  }
-};
-
-#ifdef HEPMC_PROCESS_AUX
-//--------------------------------------------------------------------
-/**
  *  Ideally, this application should work with the case where only
  *
  *  - @c o2::aod::McCollisions
  *  - @c o2::aod::McParticles
  *
- *  is available, through the use of @c
- *  o2::framework::ProcessConfigurable, but that seems to fail
- *  consistently.  The issue seems that the application @c
- *  o2-sim-mcevent-to-aod @c SIGSEGV since it stops publishing the
- *  tables when the main process of the client (this application) does
- *  not require those tables.
- *
- *  I tried various combinations of options for @c
- *  o2-sim-mcevent-to-aod but nothing seems to work.
- *
- *  The error is
- *
- *  @verbatim
- *  Exception caught: Unable to find OutputSpec with label HepMCXSections. Available Routes:
- *  - McCollisions: AOD/MCCOLLISION/0
- *  - McParticles: AOD/MCPARTICLE/1
- *  - : TFF/TFFilename/0
- *  - : TFN/TFNumber/0
- *  @endverbatim
- *
- * Or
- *
- * @verbatim
- * InputRecord::get: no input with binding HepMCHeavyIons found. Available inputs: McCollisions, McParticles
- * @endverbatim
+ * This is selected by the option `--hepmc-no-aux`
  *
- *  Interstingly, the application @c o2-sim-mcevent-to-aod works fine
- *  on its own, e.g., like
+ * The thing to remember here, is that each task process is expected
+ * to do a _complete_ job.  That is, a process _cannot_ assume that
+ * another process has been called before-hand or will be called
+ * later, for the same event in the same order.
  *
- *  @verbatim
- *  ./o2-sim-kine-publisher \
- *    --aggregate-timeframe 1 \
- *    --kineFileName pythia8pp |
- *  ./o2-sim-mcevent-to-aod \
- *    --aod-writer-keep dangling
- *  @endverbatim
+ * That is, each process will get _all_ events of a time-frame and
+ * then the next process will get _all_ events of the time-frame.
  *
- *  works fine.
+ * Processed do not process events piece-meal, but rather in whole.
  *
- * Actually, it is not likely that this will ever work.  The various
- * process are done out of sync.  That is, first all input events of
- * the timeframe are passed to the regular `process` method - i.e.,
- * tracks and collision headers are processed.  Then all input events
- * of the timeframe are passed to the optional `processAux` method -
- * i.e., auxiliary tables and collision headers.
- *
- * That means that we cannot correlate the tracks and aux tables into
- * one event, which is what we need to format a proper HepMC
- * event. The reason why it could work in the above example is because
- * we only process one timeframe at a time.
  */
-struct Task3 {
+struct Task {
```

Comment:
We have a convention in place, where the `CamelCase` task struct will be converted to `-camel-case` of the running process name. Could you please add a meaningful name here, instead of relying on `TaskName` workaround? `AodMcToHepmc` would achieve the desired name.

Reply:
It's a convention right?  Does it really matter?   How will the name of the struct be converted?  Like 

```FooBar -> o2-foo-bar```  

or just 

```FooBar -> foo-bar``` 

I think it makes sense that the process name actually matches the name of the executable file.  This relates to comment below too.

Reply:
The executable name is prefixed with `o2-<component>-`  for different reasons. There is absolutely no need to match process name with executable name, because a single executable can create many different processes.